### ISSUE TITLE
[BUG] Spark nightly cache tests fail with -- master flag

### ISSUE THREAD

**Issue by razajafri** [NVIDIA/spark-rapids#2203]
**Describe the bug**
When submitting cache tests in the integration tests [here](https://blossom.nvidia.com/sw-gpu-spark-jenkins/job/rapids_integration-dev-github/) with command 

```
spark-submit --master spark://raza-linux-2:7077 --executor-memory 12G --total-executor-cores 6 --conf spark.sql.shuffle.partitions=12 --conf spark.driver.extraClassPath=/home/rjafri/audit-plugin/jars/cudf-0.20-SNAPSHOT-cuda11.jar:/home/rjafri/audit-plugin/jars/rapids-4-spark_2.12-0.6.0-SNAPSHOT.jar:/home/rjafri/audit-plugin/jars/rapids-4-spark-udf-examples_2.12-0.6.0-SNAPSHOT.jar --conf spark.executor.extraClassPath=/home/rjafri/audit-plugin/jars/cudf-0.20-SNAPSHOT-cuda11.jar:/home/rjafri/audit-plugin/jars/rapids-4-spark_2.12-0.6.0-SNAPSHOT.jar:/home/rjafri/audit-plugin/jars/rapids-4-spark-udf-examples_2.12-0.6.0-SNAPSHOT.jar --conf spark.driver.extraJavaOptions=-Duser.timezone=UTC --conf spark.executor.extraJavaOptions=-Duser.timezone=UTC --conf spark.sql.session.timeZone=UTC --conf spark.sql.cache.serializer=com.nvidia.spark.rapids.shims.spark311.ParquetCachedBatchSerializer --jars /home/rjafri/audit-plugin/jars/rapids-4-spark-integration-tests_2.12-0.6.0-SNAPSHOT.jar ./runtests.py -v -rfExXs --std_input_path=/home/rjafri/audit-plugin/integration_tests/src/test/resources/ -k cache_test.py
```

They fail due to an exception. 

``` E                   py4j.protocol.Py4JJavaError: An error occurred while calling o282.collectToPython.
E                   : org.apache.spark.SparkException: Job aborted due to stage failure: Task 0 in stage 4.0 failed 4 times, most recent failure: Lost task 0.3 in stage 4.0 (TID 33) (192.168.1.151 executor 0): java.lang.NullPointerException
E                   	at org.apache.spark.sql.internal.SQLConf.setConfWithCheck(SQLConf.scala:3856)
E                   	at org.apache.spark.sql.internal.SQLConf.setConfString(SQLConf.scala:3701)
E                   	at com.nvidia.spark.rapids.shims.spark311.ParquetCachedBatchSerializer$CachedBatchIteratorProducer$InternalRowToCachedBatchIterator.next(ParquetCachedBatchSerializer.scala:1081)
E                   	at com.nvidia.spark.rapids.shims.spark311.ParquetCachedBatchSerializer$CachedBatchIteratorProducer$InternalRowToCachedBatchIterator.next(ParquetCachedBatchSerializer.scala:948)
E                   	at scala.collection.Iterator$$anon$10.next(Iterator.scala:459)
E                   	at org.apache.spark.storage.memory.MemoryStore.putIterator(MemoryStore.scala:222)
E                   	at org.apache.spark.storage.memory.MemoryStore.putIteratorAsValues(MemoryStore.scala:299)
```
Removing the `-- master spark://aaa:bbb` flag from the command makes the script pass

**Steps/Code to reproduce bug**
set WORKSPACE to a local tmp folder and set `URM_URL=https://urm.nvidia.com/artifactory/sw-spark-maven`
run jenkins/spark-tests

**Expected behavior**
All tests should pass

**Comment by tgravescs** [comment]
we need to at least test the caching in distributed enivronment before 0.5. If its just a test issue that is different, if user can't use then we need to know that.

**Comment by razajafri** [comment]
This is fixed as part of #2227
